### PR TITLE
Payload compression

### PR DESCRIPTION
**Pull request by gszol**
Simple payload compression sample. Copy-paste of the prov-deprov sample.

### CODE REVIEW

File: src/features/PayloadCompression.js
Comment (Mrc0113) [inline src/features/PayloadCompression.js:65]: what does "9" mean? can we include a detailed comment with context?

**Review by Mrc0113 — CHANGES_REQUESTED** [review]

File: src/features/PayloadCompression.js
Comment (gszol) [reply src/features/PayloadCompression.js:65]: sure, one sec.

Comment (gszol) [reply src/features/PayloadCompression.js:70]: Done.

**Comment by gvensan** [comment]
Just curious - I maybe wrong. Compression is available only for SMF. How do we test whether the payload is getting compressed or not (maybe a before and after length log would help)

As a sample run does not indicate a clear impact of compression: 

```
Session subscribed to topic, sending message with a thousand bytes, all the same
Message received. Payload length:1000
TX bytes:1048
RX bytes:1048
```

File: src/features/PayloadCompression.js
Comment (gvensan) [inline src/features/PayloadCompression.js:1]: Tested, working great!

**Comment by gvensan** [comment]
Hi @gszol - When is this planned to be released?

**Comment by gszol** [comment]
> Hi @gszol - When is this planned to be released?

Looks like we just released it with 10.17.0 : https://products.solace.com/download/JS_API_RN

**Comment by Mrc0113** [comment]
@gszol if this is ready to be merged can you change it from a draft PR?

**Comment by gszol** [comment]
> @gszol if this is ready to be merged can you change it from a draft PR?

I though I have, sorry.
Now I again think I just did, I guess we'll see.

**Comment by gvensan** [comment]
Is there anything blocking this PR? Can we merge it.....

**Comment by gszol** [comment]
> Is there anything blocking this PR? 

Not to my knowledge.

> Can we merge it.....

Yes, please do. (I don't have write access.)

**Pull request merged**
